src/test/kotlin/io/bazel/worker/WorkerEnvironment.kt

```diff
@@ -43,12 +43,17 @@ object WorkerEnvironment {
       PipedStream().use { stdOut ->
         PipedStream().use { stdIn ->
           val responses = ConcurrentLinkedQueue<WorkResponse>()
-          executor.submit {
-            generateSequence { WorkResponse.parseDelimitedFrom(stdOut.input) }.forEach {
-              responses.add(it)
+          val reader = executor.submit {
+            Thread.currentThread().name = "readStdOut"
+            while (true) {
+              WorkResponse.parseDelimitedFrom(stdOut.input)
+                ?.let(responses::add)
+                ?: break
```

Comment:
Ok, I like this sneaky little pattern.

Reply:
It's very jetbrains kotlin. Been reading too much code from there.